### ISSUE TITLE
Fit BarqCurve control points to curve

### ISSUE THREAD

**Issue by gderossi** [error-corp/qurveros#5]
Currently, the control points used by BarqCurve objects can be initialized in two ways. Users can either pass an array of 3D points to [BarqCurve.initialize_parameters](https://github.com/error-corp/qurveros/blob/e00753b2187fa19c9af02e03375e6440e320fb79/src/qurveros/optspacecurve.py#L277) via the `init_free_points` keyword argument, or control points will be generated randomly, with an optional seed. 

```python
if init_free_points is None:
    init_free_points = rng.standard_normal((self.n_free_points, 3))
    init_free_points = \
        init_free_points/numpy.linalg.norm(init_free_points, axis=0)

    init_free_points = jnp.array(init_free_points)

else:
    if init_free_points.shape[0] != self.n_free_points:
        raise ValueError('Inconsistent number of free points with'
                         ' provided initial free points.')

params['free_points'] = init_free_points
return super().initialize_parameters(params)
```

Starting these control points at an informed initial condition instead of a random one could greatly improve optimization performance. We would like to fit control points to a curve defined either by a function or an array of points. More accurately, we would like to fit a Bezier curve defined by control points to this input curve, which is a bit more challenging than just interpolating N points along the curve and returning them as control points.

Curve fitting could be implemented either as a refactor of `BarqCurve.initialize_parameters` or as a new function that returns an array of control points, which would then be passed to `BarqCurve.initialize_parameters`. Note that the current function signature only accepts values for the free control points, not all of the control points (which include a few fixed points on either end), so a small rework to `BarqCurve.initialize_parameters` will likely be required if creating a new curve fitting function.

**Comment by Kitsunp** [comment]
You can something like assign me to this issue please.

**Comment by gderossi** [comment]
Hi @Kitsunp, thanks for working on this and issue #6! We've been a bit busy the past few days, but we are reviewing your PRs and will provide feedback by the end of tomorrow!

**Comment by gderossi** [comment]
Closed by #14
